spec/feature/posts_index_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe 'User post index page', type: :feature do
+  let!(:user) { User.create(name: 'Sarkodie', photo: '/path/to/photo1.jpg', post_counter: 10) }
+
+  before do
+    5.times do
+      Post.create(
+        title: 'Post title',
+        text: 'Post text',
+        author_id: user.id,
+        comments_counter: 3,
+        likes_counter: 7
+      )
+    end
+    visit user_posts_path(user)
+  end
+
+  it 'displays user profile information' do
+    expect(page).to have_css("img[src='#{user.photo}'][alt='user photo']")
+    expect(page).to have_content(user.name)
+    expect(page).to have_content("Number of posts: #{user.post_counter}")
+  end
+
+  it 'displays each post with relevant information' do
+    @posts = user.posts
+
+    @posts.each do |post|
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.text[0..49])
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+      expect(page).to have_content("Likes: #{post.likes_counter}")
+      expect(page).to have_link('Read post', href: user_post_path(post))
+    end
+  end
```

Comment:
- [ ] As suggested by the previous reviewer, kindly consider rewriting this test by first `click_link('id-of-link')` and then use `have_current_path` matcher like following.
`# Use the `have_current_path` matcher to confirm the current page
    expect(page).to have_current_path(post_path(post))`

Reply:
Hi @Zuhaib042!

I have already added a separate test case to handle this situation.

![Capture](https://github.com/CaptainBawa/Blog-App/assets/82339317/35d965f5-caf6-4f95-85f7-17a93e810fb7)